### ISSUE TITLE
jfs mounted on Windows way slower than jfs mounted on Linux exposed via smb

### ISSUE THREAD

**Issue by happyc0ding** [juicedata/juicefs#5252]
**What happened**:
I have 1 client (Windows 10), 1 server (Linux).
(Background: Mounting an image via Arsenal Image Mounter)
Scenario 1:
Mounting jfs in Windows via network and opening a big file (~150GB): Takes about 60 seconds.
Scenario 2:
Mounting the same jfs on the Linux server directly and exposing the mount via smb, then mounting this share via the same network in Windows and opening the same file: Takes about 12 seconds.

Playing around with prefetch, cache-partial-only and setting cache-dir to memory did not change much (caching might have been an explanation, since the local disk is slower than the disk minio is running on).
I've tried several times in order to confirm the timing and ruling out any other caching mechanisms.

**What you expected to happen**:
I expected the performance in scenario 1 and 2 to be roughly the same or scenario 2 to be slower since it adds another protocol layer.

**How to reproduce it (as minimally and precisely as possible)**:
Install jfs on Windows, install jfs, samba, minio and mariadb on linux. Create a jfs and mount it.
Mounted on linux as described in the docs: `sudo juicefs mount -d --enable-xattr mysql://... /mnt/myjfs` and used a  default smb.conf provided by the package, only added a share as described in [the docs](https://juicefs.com/docs/community/deployment/samba/#step-3-create-a-samba-share).
Mounted jfs in windows with `juicefs.exe mount "mysql://..." x:`
Mounted smb in windows with `net use x: \\foo\bar`.

**Anything else we need to know?**
Not sure if this is a bug. However, I did not find anything in the docs explaining this. Is the windows driver that much slower than linux? If yes, it would be nice to know.

**Environment**:
- JuiceFS version (use `juicefs --version`) or Hadoop Java SDK version: juicefs version 1.1.4+2024-08-30.fc080fd on windows, juicefs version 1.2.1+2024-08-30.cd871d1 on linux
- OS (e.g `cat /etc/os-release`): Windows 10 / Debian 12
- Kernel (e.g. `uname -a`): Linux 6.1.0-26-amd64
- Object storage (cloud provider and region, or self maintained): self hosted minio version RELEASE.2024-10-02T17-50-41Z
- Metadata engine info (version, cloud provider managed or self maintained): mariadb from 11.5.2-MariaDB
- Network connectivity (JuiceFS to metadata engine, JuiceFS to object storage): 10G via Vmware
- Others: Using winfsp-2.0.23075

**Comment by chenjie4255** [comment]
Here are some additional tests I did, using the latest main branch of juicefs:

* Run minio, redis on a linux system by vmware (4 core, 8g) 
* Use vmware nat forward to expose redis port and minio port.
* Install wsl2 ubuntu 20.04, mount juicefs with default parameters, and run juicefs bench (also with default parameters)
* Install winfsp, and mount juicefs on the host machine, then run the juicefs.exe bench command.
* (All the softwares are actually on a same machine)
* Hardware: AMD 7735 HS,  64g ram.

```
WSL version: 2.4.13.0
Kernel version: 5.15.167.4-1
WSLg version: 1.0.65
MSRDC version: 1.2.5716
Direct3D version: 1.611.1-81528511
DXCore version: 10.0.26100.1-240331-1435.ge-release
Windows version: 10.0.22631.4751
```

The tests showed that the performance between big file reading and writing are basically the same.
As for the small file and stats tests, the WSL has a significantly higher performance than the Windows version.

(The left side of the screenshots shows the juicefs bench result from the WSL)

![Image](https://github.com/user-attachments/assets/d9a675ac-221c-4cda-91b0-d32a531ed7a2)
![Image](https://github.com/user-attachments/assets/e09361c5-c6a7-4163-be79-52aec831f8f5)
![Image](https://github.com/user-attachments/assets/64056bfd-36f8-4826-8784-590ad8a17ec1)
![Image](https://github.com/user-attachments/assets/95b17b09-5dd5-49d8-8709-ee09054a6a6b)

 The big file performance is abnormally low, maybe it is due to the VMware nat forward or some other settings on my PC.

cc @davies

**Comment by chenjie4255** [comment]
Oh you reminded me, my disk usage had exceeded 90% and I ingored the startup disk warning.

Reran the test with free-space-ratio 0.01, here are the results:

![Image](https://github.com/user-attachments/assets/30955dab-764b-445b-9e6f-cb1c866f9c16)
![Image](https://github.com/user-attachments/assets/0b429589-b77e-48c6-864a-e9976bdec257)
![Image](https://github.com/user-attachments/assets/3bfeeb85-55b6-46b1-b6fa-5e5432d1dc9b)

The small file reading performance has improved but is still far behind the WSL.

**Comment by davies** [comment]
I think the difference came from path translation (path or inode). When MariaDB is used, the difference could be bigger.

Can you check the number of meta operation, such as lookup and getattr?

**Comment by chenjie4255** [comment]
> I think the difference came from path translation (path or inode). When MariaDB is used, the difference could be bigger.
> 
> Can you check the number of meta operation, such as lookup and getattr?

Of course, here are the comparisons from the access log, captured during the bench test. 

![Image](https://github.com/user-attachments/assets/e7cc76b9-40fe-4b53-9551-a6af97eb596e)

WSL:

* GetAttr:241
* lookup: 312
* write:8293
* total accesslog line: 17890

Windows:
* GetAttr: 2750
* lookup: 0
* write: 1124
* total lines of access log: 6752

[wsl_bench.zip](https://github.com/user-attachments/files/19697848/wsl_bench.zip)
